### ISSUE TITLE
Error occurred while executing task RescanFolders: Length cannot be less than zero. (Parameter 'length')

### ISSUE THREAD

**Issue by kamtschatka** [Readarr/Readarr#1473]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I was scanning a folder with a lot of books in it, to import them.
The import failed and showed this exception:

Error occurred while executing task RescanFolders: Length cannot be less than zero. (Parameter 'length')

```
System.ArgumentOutOfRangeException: Length cannot be less than zero. (Parameter 'length')
   at System.String.Substring(Int32 startIndex, Int32 length)
   at NzbDrone.Core.Parser.Parser.SplitBookTitle(String book, String author) in D:\a\1\s\src\NzbDrone.Core\Parser\Parser.cs:line 591
   at NzbDrone.Core.MediaFiles.BookImport.Identification.DistanceCalculator.BookDistance(List`1 localTracks, Edition edition) in D:\a\1\s\src\NzbDrone.Core\MediaFiles\BookImport\Identification\DistanceCalculator.cs:line 27
   at NzbDrone.Core.MediaFiles.BookImport.Identification.IdentificationService.GetBestRelease(LocalEdition localBookRelease, IEnumerable`1 candidateReleases, List`1 extraTracksOnDisk, Boolean& seenCandidate) in D:\a\1\s\src\NzbDrone.Core\MediaFiles\BookImport\Identification\IdentificationService.cs:line 205
   at NzbDrone.Core.MediaFiles.BookImport.Identification.IdentificationService.IdentifyRelease(LocalEdition localBookRelease, IdentificationOverrides idOverrides, ImportDecisionMakerConfig config) in D:\a\1\s\src\NzbDrone.Core\MediaFiles\BookImport\Identification\IdentificationService.cs:line 145
   at NzbDrone.Core.MediaFiles.BookImport.Identification.IdentificationService.Identify(List`1 localTracks, IdentificationOverrides idOverrides, ImportDecisionMakerConfig config) in D:\a\1\s\src\NzbDrone.Core\MediaFiles\BookImport\Identification\IdentificationService.cs:line 87
   at NzbDrone.Core.MediaFiles.BookImport.ImportDecisionMaker.GetImportDecisions(List`1 musicFiles, IdentificationOverrides idOverrides, ImportDecisionMakerInfo itemInfo, ImportDecisionMakerConfig config) in D:\a\1\s\src\NzbDrone.Core\MediaFiles\BookImport\ImportDecisionMaker.cs:line 157
   at NzbDrone.Core.MediaFiles.DiskScanService.Scan(List`1 folders, FilterFilesType filter, Boolean addNewAuthors, List`1 authorIds) in D:\a\1\s\src\NzbDrone.Core\MediaFiles\DiskScanService.cs:line 164
   at NzbDrone.Core.MediaFiles.DiskScanService.Execute(RescanFoldersCommand message) in D:\a\1\s\src\NzbDrone.Core\MediaFiles\DiskScanService.cs:line 313
   at NzbDrone.Core.Messaging.Commands.CommandExecutor.ExecuteCommand[TCommand](TCommand command, CommandModel commandModel) in D:\a\1\s\src\NzbDrone.Core\Messaging\Commands\CommandExecutor.cs:line 111
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid3[T0,T1,T2](CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at NzbDrone.Core.Messaging.Commands.CommandExecutor.ExecuteCommands() in D:\a\1\s\src\NzbDrone.Core\Messaging\Commands\CommandExecutor.cs:line 42
```

### Expected Behavior

No exception is thrown

### Steps To Reproduce

1. Set a root foldeer
2. Rescan the folder

### Environment

```markdown
- OS: Unraid
- Readarr: 0.1.0.1209
- Docker Install: yes
- Using Reverse Proxy: no
- Browser: Chrome 97
```

### What branch are you running?

Nightly

### Trace Logs?

I didn't have it running in trace mode as it always fills up my docker disk and causes problems. Rerunning is not an option on the weekend as the full import takes about a day, fills up all 32GB of RAM on my NAS and renders it basically useless.

I however tracked down the issue and created a sample test:

This goes into the ParserFixture.cs file: 
```
        [Test]
        public void should_handle_parenthesis()
        {
            Parser.Parser.SplitBookTitle("asdf)(", "");
        }
```

Basically the closing parenthesis is before the opening one. I am guessing adding a startindex for the indexOf call is enough, but I am not sure.

[AB#2424](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/2424)

**Comment by ta264** [comment]
For the record trace logs are limited to 50mb. 

I'll take a look, thanks.

Please do join the discord channel though, I much prefer working through issues in real time
